### ISSUE TITLE
none unique anchors 

### ISSUE THREAD

**Issue by b4ldr** [puppetlabs/puppet-strings#341]
## Describe the Bug
While adding some [type definitions to stdlib](https://github.com/puppetlabs/puppetlabs-stdlib/pull/1299#discussion_r1132464346) it was noticed that puppet-strings generates the same anchore for thge following two types

```puppet
Stdlib::Httpstatus
Stdlib::Http::Status
```

This seems to be because strings does something equivalent to the following

```ruby
anchor =- @type.tr(':', '').downcase
```

i think it would be better to replace the '::' with something like '__' to keep the anchors unique

## Expected Behaviour
All types should have unique anchors

## Steps to Reproduce
create the following two types and run puppet strings

```puppet
@summary some documentation 
Stdlib::Httpstatus = String
@summary some other documentation 
Stdlib::Http::Status = String
```

**Comment by ekohl** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a states `name` is deprecated:
> Was required to define a possible target location in a page. In HTML 4.01, id and name could both be used on <a>, as long as they had identical values.
>
> Note: Use the global attribute [id](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#id) instead.

Then https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id states:
> Note: Technically, the value for an id attribute may contain any character, except [whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) characters. However, to avoid inadvertent errors, only [ASCII](https://developer.mozilla.org/en-US/docs/Glossary/ASCII) letters, digits, '_', and '-' should be used and the value for an id attribute should start with a letter. For example, . has a special meaning in CSS (it acts as a [class selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors)). Unless you are careful to escape it in the CSS, it won't be recognized as part of the value of an id attribute. It is easy to forget to do this, resulting in bugs in your code that could be hard to detect.

Given `-` is invalid in a Puppet type alias I'd recommend to use that instead of `:`.

**Comment by b4ldr** [comment]
this is a duplicate of #300 and fixed in #303 @ekohl should we create a separate bug for the id vs name issue
